observability/src/main/java/io/grpc/observability/GlobalLoggingTags.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.observability;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.auth.http.HttpTransportFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableMap;
+import io.grpc.observability.metadata.MetadataConfig;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.function.Function;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/** A container of all global custom tags used for logging (for now). */
+final class GlobalLoggingTags {
+  private static final Logger logger = Logger.getLogger(GlobalLoggingTags.class.getName());
+
+  private static final String ENV_KEY_PREFIX = "GRPC_OBSERVABILITY_";
+  private final Map<String, String> tags;
+
+  GlobalLoggingTags() {
```

Comment:
Package private for a reason? This will only be used from this package?

Reply:
Correct. When properly wired up, this will only be used by `Observability` class which will get the tags and inject them into the interceptor factories.